### PR TITLE
Add longer timeout for dotnet-watch test

### PR DESCRIPTION
**Pull request by BrennanConroy**
@muratg Test only change.

Also, realized the retry doesn't work for this test because the DOTNET_WATCH_ITERATION count can increase twice if it retries.

Fixes https://github.com/aspnet/AspNetCore-Internal/issues/1696

### CODE REVIEW

**Review by muratg — COMMENTED** [review]
Per @natemcmaster there's another fix in 3.0, and the timeout would not actually work. Let's see if we can find it.

**Comment by natemcmaster** [comment]
This is the flaky test fix I was trying to find that I don't think made it into 2.2: https://github.com/aspnet/DotNetTools/commit/a24b4ee4598b425d9e8b2f041be508c32eb41cf5

**Comment by BrennanConroy** [comment]
So it's not related

**Review by muratg — APPROVED** [review]

**Comment by natemcmaster** [comment]
If it fixes it, :shipit:

**Comment by BrennanConroy** [comment]
Am I the only one hitting build timeouts in our 2.2 builds? @natemcmaster @Eilon Should we increase the AzDo timeouts or is there a better solution?

**Comment by natemcmaster** [comment]
Based on a cursory inspection of the log, I think we may be running into known issues which deadlock EFCore tests on AzDO. Increasing the timeout would not help.

I would be in favor of skipping EFCore tests completely when building it as a submodule of AspNetCore. The EFCore repo already run tests on all pull requests to 2.x servicing branches.

@ajcvickers @Eilon  would that be okay with you?

**Comment by Eilon** [comment]
If that logic is correct, I'm fine with it.

**Review by muratg — APPROVED** [review]

File: build/RepositoryBuild.targets [hunk 1]
```diff
@@ -56,7 +57,8 @@
       StopOnFirstFailure="false"
       Targets="_TestRepository"
       Properties="BuildNumber=$(BuildNumber);IsFinalBuild=$(IsFinalBuild);Configuration=$(Configuration);_NoBuildRepos=$(_NoBuildRepos)"
-      ContinueOnError="true">
+      ContinueOnError="true"
+      Condition=" @(_BatchedTestRepo.SkipTests) != 'true' ">
```
Comment (BrennanConroy) [inline build/RepositoryBuild.targets]: @natemcmaster This doesn't seem to do anything

Comment (natemcmaster) [reply build/RepositoryBuild.targets]:
```suggestion
      Condition=" '%(_BatchedTestRepo.SkipTests)' != 'true' ">
```

Might be due to wrong syntax...

Comment (BrennanConroy) [reply build/RepositoryBuild.targets]: Oh, the quotes...

**Comment by BrennanConroy** [comment]
@natemcmaster FYI I moved the msbuild check and it finally worked.

**Comment by natemcmaster** [comment]
🎉 thanks for figuring this one out.

**Pull request merged**